core-api/src/main/java/com/optimizely/ab/config/audience/NotCondition.java

```diff
@@ -37,7 +37,7 @@ public Condition getCondition() {
         return condition;
     }
 
-    public boolean evaluate(Map<String, String> attributes) {
+    public boolean evaluate(Map<String, ?> attributes) {
```

Comment:
Do we need to update the `toString` method on this class to account for non-string values? Seems like it could lead to exceptions